### DISCUSSION TITLE
how to configure hnsw index via WeaviateDocumentIndex

### DISCUSSION THREAD

**Discussion by mmoya91** [docarray/docarray#1902] [Q&A]
Hello, there are some configurations I'd like to consider given weaviate's hnsw index
https://weaviate.io/developers/weaviate/config-refs/schema/vector-index#hnsw-index-parameters
I was wondering how could I consider some of these parameters given `WeaviateDocumentIndex`?

**Comment by JoanFM** [comment]
Have you read the documentation here?

https://docs.docarray.org/user_guide/storing/index_weaviate/

specially those about Configuration and RuntimeConfiguration

**Reply by mmoya91** [reply]
briefly yes, but looking at the source code via weaviate
https://weaviate-python-client.readthedocs.io/en/v3.2.2/weaviate.schema.html#module-weaviate.schema

and docarray, it looks like I would have to modify the source code here
https://github.com/docarray/docarray/blob/75e0033a361a31280709899e94d6f5e14ff4b8ae/docarray/index/backends/weaviate.py#L221-L227
and add something like
​`schema["vectorIndexConfig"] = self._vector_index_config `

if I wanted to modify something like [vectorCacheMaxObjects](https://weaviate.io/developers/weaviate/config-refs/schema/vector-index#hnsw-index-parameters)

which matches up with this TODO: https://github.com/docarray/docarray/blob/75e0033a361a31280709899e94d6f5e14ff4b8ae/docarray/index/backends/weaviate.py#L221-L222
it would be nice to add other configs like `vectorIndexConfig`. I currently need to consider a vector cache because weaviate's hnsw index is an [in-memory](https://weaviate.io/developers/weaviate/concepts/resources#the-role-of-memory) index but the only way to consider a vector cache is to alter the `vectorIndexConfig` upon schema creation which looks like something that  docarray does not yet support
